include/alp/algorithms/cholesky.hpp

```diff
@@ -233,26 +228,29 @@ namespace alp {
 #endif
 
 			//nb: number of blocks of (max) size bz
-			size_t nb = n / bs ;
+			if( ( bs == 0 ) || ( bs > n ) ) {
+				std::cerr << "Block size has illegal value, bs =   " << bs << " .\n";
+				std::cerr << "It should be from interval < 0,  " << n << "] .\n";
+				return FAILED;
+			}
+			size_t nb = n / bs;
 			if( n % bs != 0 ){
-				nb = nb + 1 ;
+				nb = nb + 1;
 			}
 
 
-			for( size_t i = 0; i < nb ; ++i ) {
-				size_t a = std::min( i * bs, n );
-				size_t b = std::min( ( i + 1 ) * bs, n );
-				size_t c = n;
+			for( size_t i = 0; i < nb; ++i ) {
+				const size_t a = std::min( i * bs, n );
+				const size_t b = std::min( ( i + 1 ) * bs, n );
+				const size_t c = n;
 
-				auto range1 = utils::range( a, b );
-				auto range2 = utils::range( b, c );
+				const utils::range range1 = utils::range( a, b );
```

Comment:
Consider writing like `const utils::range range1( a, b )`. Like this the variable is initialized, rather than copy constructed.
Same comment for the line below.